src/Externals/spire/arc-look-at/ArcLookAt.cpp

```diff
@@ -103,10 +104,12 @@ void ArcLookAt::doZoom(const glm::vec2& ssPos)
 //------------------------------------------------------------------------------
 glm::mat4 ArcLookAt::getWorldViewTransform() const
 {
-  glm::mat4 finalTrafo;
-  finalTrafo[3].xyz() = -mCamLookAt;
-  finalTrafo          = mArcBall->getTransformation() * finalTrafo;
-  finalTrafo[3][2]   += -mCamDistance;
+  glm::mat4 finalTrafo(1.0);
+  for (int i = 0; i < mCamLookAt.length(); ++i)
+    finalTrafo[3][i] = -mCamLookAt[i];
+
+  finalTrafo = mArcBall->getTransformation() * finalTrafo;
+  finalTrafo[3][2] -= mCamDistance;
```

Comment:
Can this be unit tested separately? The camera rotation problems are probably coming from here.

Reply:
I added a test that tests these transformations: offset->rotate->offset. This does not seem to be where the problems are coming from.